Library/Homebrew/extend/os/mac/keg_relocate.rb

```diff
@@ -41,9 +41,9 @@ def relocate_dynamic_linkage(relocation)
 
         if ENV["HOMEBREW_RELOCATE_RPATHS"]
           each_rpath_for(file) do |old_name|
-            new_name = if old_name.start_with? old_cellar
+            new_name = if relocation.start_with_old_value? :cellar, old_name
               old_name.sub(old_cellar, new_cellar)
-            elsif old_name.start_with? old_prefix
+            elsif relocation.start_with_old_value? :prefix, old_name
```

Comment:
Relocation was supposed to happen here but didn't:
```
❯ otool -l * | rg -A2 LC_RPATH
          cmd LC_RPATH
      cmdsize 40
         path /usr/local/opt/llvm/lib (offset 12)
--
          cmd LC_RPATH
      cmdsize 32
         path @loader_path/ (offset 12)
--
          cmd LC_RPATH
      cmdsize 40
         path /usr/local/opt/llvm/lib (offset 12)
--
          cmd LC_RPATH
      cmdsize 32
         path @loader_path/ (offset 12)
```

Reply:
What's supposed to happen here?

When testing on `master` I get the same output

Reply:
This is the correct output:
```
❯ otool -l * | rg -A2 LC_RPATH
          cmd LC_RPATH
      cmdsize 48
         path @@HOMEBREW_PREFIX@@/opt/llvm/lib (offset 12)
--
          cmd LC_RPATH
      cmdsize 32
         path @loader_path/ (offset 12)
--
          cmd LC_RPATH
      cmdsize 48
         path @@HOMEBREW_PREFIX@@/opt/llvm/lib (offset 12)
--
          cmd LC_RPATH
      cmdsize 32
         path @loader_path/ (offset 12)
```